lib/features/modeling/behavior/EventBasedGatewayBehavior.js

```diff
@@ -0,0 +1,73 @@
+import inherits from 'inherits';
+
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+import { is } from '../../../util/ModelUtil';
+
+export default function EventBasedGatewayBehavior(eventBus, modeling) {
+
+  CommandInterceptor.call(this, eventBus);
+
+  /**
+   * Remove existing sequence flows of event-based target before connecting
+   * from event-based gateway.
+   */
+  this.preExecuted('connection.create', function(event) {
+
+    var source = event.context.source,
+        target = event.context.target,
+        existingIncomingConnections = target.incoming.slice();
+
+    if (
+      is(source, 'bpmn:EventBasedGateway') &&
+      target.incoming.length > 0
+    ) {
+
+      existingIncomingConnections
+        .filter(function(connection) {
+          return is(connection, 'bpmn:SequenceFlow');
+        })
+        .forEach(function(sequenceFlow) {
+          modeling.removeConnection(sequenceFlow);
+        });
+    }
+  });
+
+  /**
+   *  After replacing shape with event-based gateway, remove incoming sequence
+   *  flows of event-based targets which do not belong to event-based gateway
+   *  source.
+   */
+  this.preExecuted('shape.replace', function(event) {
+
+    var newShape = event.context.newShape,
+        newShapeOutgoingConnections = newShape.outgoing;
+
+    if (!is(newShape, 'bpmn:EventBasedGateway')) {
+      return;
+    }
+
+    newShapeOutgoingConnections
```

Comment:
This part should be refactored as it's really difficult to understand. For sure we don't need `map` here. Consider assigning step results to variables.